engine/dlib/src/dlib/configfile.cpp

```diff
@@ -255,6 +255,20 @@ namespace dmConfigFile
         buf[i] = '\0';
     }
 
+    static int SizeOfLiteral(Context* context)
+    {
+        int pos = context->m_BufferPos;
+        int c = GetChar(context);
+        int i = 0;
+        while (c != '\n' && c != '\r')
+        {
+            c = GetChar(context);
+            ++i;
+        }
+        context->m_BufferPos = pos;
+        return i;
+    }
```

Comment:
A common api (for various C functions) is to use the same function, but not pass in the output buffer:

```
static int ParseLiteral(Context* context, char* buf, int buf_len)
{
    int c = GetChar(context);
    int i = 0;
    while (c != '\n' && c != '\r')
    {
        if (buf != 0)
        {
            buf[i] = c;
            if (i >= (int) buf_len - 1)
                ParseError(context, RESULT_LITERAL_TOO_LONG);
        }
        c = GetChar(context);
        ++i;
    }
    BufferUngetChar(c, context);
    if (buf != 0)
        buf[i] = '\0';
    return i;
}
```

Reply:
I considered this but felt that the ParseLiteral function became a bit too messy:

```
    static int ParseLiteral(Context* context, char* buf, int buf_len)
    {
        int pos = context->m_BufferPos;
        int c = GetChar(context);
        int i = 0;
        while (c != '\n' && c != '\r')
        {
            if (buf != 0)
            {
                buf[i] = c;
                if (i >= (int) buf_len - 1)
                    ParseError(context, RESULT_LITERAL_TOO_LONG);
            }
            c = GetChar(context);
            ++i;
        }
        if (buf != 0)
        {
            BufferUngetChar(c, context);
            buf[i] = '\0';
        }
        else
        {
            context->m_BufferPos = pos;
        }
        return i;
    }
```

Reply:
The downside to having two functions is that the iterator functionality is also split into two functions.
It's easier for the implementations to diverge.

Reply:
That is a good point. I'll use the combined function instead.